samples/cpp/shape_example.cpp

```diff
@@ -54,17 +54,14 @@ static vector<Point> simpleContour( const Mat& currentQuery, int n=300 )
 
 int main(int argc, char** argv)
 {
-    help();
     string path = "../data/shape_sample/";
-    int indexQuery = 1;
-    if( argc < 2 )
+    cv::CommandLineParser parser(argc, argv, "{help h||}{@input|1|}");
+    if (parser.has("help"))
     {
-        std::cout<<"Using first image as query."<<std::endl;
-    }
-    else
-    {
-        sscanf( argv[1], "%i", &indexQuery );
+        help();
+        return 0;
     }
+    int indexQuery = parser.get<int>("@input");
```

Comment:
We can also check whether the `indexQuery` value is in mentioned range (`[1;20]`).